### PR TITLE
Create annotation model and endpoints

### PR DESCRIPTION
**Pull request by furkanakkurt1335**
- create annotation models (annotation, body, target and selector)
- add get annotation endpoint

### CODE REVIEW

File: app/server/src/models/annotation/body.model.js [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+const createBody = async (purpose, type, value) => {
+  var body = new Body({
+    purpose,
+    type,
+    value,
+  });
+  body.created = body.createdAt;
+  const res = await body.save();
+  return res;
```
Comment (ersoykadir) [inline app/server/src/models/annotation/body.model.js:36]: Where do we create `creator` field?

File: app/server/src/routes/annotation/annotation.controller.js [hunk 2]
```diff
@@ -0,0 +1,51 @@
...
+      })
+        .populate({
+          path: "body",
+          populate: { path: "creator", select: { name: 1 } },
+        })
+        .populate({
+          path: "target",
+          populate: {
+            path: "selector",
+            model: Selector,
```
Comment (ersoykadir) [inline app/server/src/routes/annotation/annotation.controller.js:35]: What does model do?

File: app/server/src/models/annotation/annotation.model.js [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+  var bodies_l = [];
+  for (var body_el of body) {
+    var body_t = new Body({
+      purpose: body_el.purpose,
+      value: body_el.value,
+      type: body_el.type,
+      creator: body_el.creator,
+      created: body_el.created,
+    });
+    await body_t.save();
```
Comment (ersoykadir) [inline app/server/src/models/annotation/annotation.model.js:44]: can you please use createBody function of Body.

File: app/server/src/models/annotation/body.model.js (hunk 1 above)
Comment (ersoykadir) [reply app/server/src/models/annotation/body.model.js:36]: Also I am not sure whether we should user createdAt to assign body.created or not. I think created will be given as input

File: app/server/src/models/annotation/annotation.model.js (hunk 3 above)
Comment (ersoykadir) [reply app/server/src/models/annotation/annotation.model.js:44]: Be aware that it is missing creator field and created field

File: app/server/src/models/annotation/annotation.model.js [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+      var selector_t = new Selector({
+        type: selector_el.type,
+        exact: selector_el.exact,
+      });
+    } else if (selector_keys.includes('start')) {
+      var selector_t = new Selector({
+        type: selector_el.type,
+        start: selector_el.start,
+        end: selector_el.end,
+      });
```
Comment (ersoykadir) [inline app/server/src/models/annotation/annotation.model.js:60]: can you please use create textposition function of Selector.

File: app/server/src/models/annotation/annotation.model.js [hunk 5]
```diff
@@ -0,0 +1,89 @@
...
+    bodies_l.push(body_t);
+  }
+  var selectors_l = [];
+  for (var selector_el of target.selector) {
+    selector_keys = Object.keys(selector_el);
+    if (selector_keys.includes('exact')) {
+      var selector_t = new Selector({
+        type: selector_el.type,
+        exact: selector_el.exact,
+      });
```
Comment (ersoykadir) [inline app/server/src/models/annotation/annotation.model.js:54]: can you please use create textquote function of Selector.

File: app/server/src/routes/annotation/annotation.controller.js [hunk 6]
```diff
@@ -0,0 +1,51 @@
...
+      return res.status(201).json({ annotation });
+    } catch (e) {
+      return res.status(400).send({ error: e.toString() });
+    }
+  },
+  getAnnotation: async function (req, res) {
+    try {
+      const { id } = req.params;
+      console.log(id);
+      const annotation = await AnnotationModel.Annotation.find({
```
[the remark this answers is not in the document]
Comment (furkanakkurt1335) [reply app/server/src/routes/annotation/annotation.controller.js:24]: Done!

File: app/server/src/routes/annotation/annotation.controller.js (hunk 2 above)
Comment (furkanakkurt1335) [reply app/server/src/routes/annotation/annotation.controller.js:35]: uses mongoose model Selector to populate

File: app/server/src/models/annotation/target.model.js [hunk 7]
```diff
@@ -0,0 +1,31 @@
+const mongoose = require("mongoose");
+
+const targetSchema = new mongoose.Schema({
+  selector: [
+    {
+      type: mongoose.Schema.Types.ObjectId,
+      ref: "Target",
```
[the remark this answers is not in the document]
Comment (furkanakkurt1335) [reply app/server/src/models/annotation/target.model.js:7]: Done!

File: app/server/src/models/annotation/annotation.model.js (hunk 5 above)
Comment (furkanakkurt1335) [reply app/server/src/models/annotation/annotation.model.js:54]: Done!

File: app/server/src/models/annotation/annotation.model.js (hunk 4 above)
Comment (furkanakkurt1335) [reply app/server/src/models/annotation/annotation.model.js:60]: Done!

File: app/server/src/models/annotation/annotation.model.js (hunk 3 above)
Comment (furkanakkurt1335) [reply app/server/src/models/annotation/annotation.model.js:44]: Done!

File: app/server/src/models/annotation/body.model.js (hunk 1 above)
Comment (furkanakkurt1335) [reply app/server/src/models/annotation/body.model.js:36]: Added!

**Comment by furkanakkurt1335** [comment]
Changed according to @ersoykadir's review. Thank you, it's very helpful!

**Review by ersoykadir — APPROVED** [review]
LGTM

**Pull request merged**